source/features/link-to-changelog-file.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import {BookIcon} from '@primer/octicons-react';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import {buildRepoURL, getRepo} from '../github-helpers';
+
+const getCacheKey = (): string => `changelog:${getRepo()!.nameWithOwner}`;
+
+function parseFromDom(): string | false {
+	return select('.js-navigation-item [title^="changelog." i]')?.textContent ?? false;
+}
+
+async function fetchFromApi(): Promise<string | false > {
+	const {repository} = await api.v4(`
+		repository() {
+			object(expression: "HEAD:") {
+				...on Tree {
+					entries {
+						name
+					}
+				}
+			}
+		}
+	`);
+
+	return repository.object.entries.find((file: {name: string}) => file.name.toLowerCase().startsWith('changelog.'))?.name ?? false;
```

Comment:
A bit clearer 

```suggestion
	for (const file of repository.object.entries) {
		if (file.name.toLowerCase().startsWith('changelog.')) {
			return file.name;
		}
	}

	return false;
```

Reply:
Found a nice extension-less changelog https://github.com/genesi/gtk2/blob/master/ChangeLog

Reply:
The issue is what if someone has a change log folder? (or maybe that a good thing to catch that)